src/CommonDBTM.php

```diff
@@ -1746,6 +1746,11 @@ protected function cleanLockedsOnAdd()
                 if ($lock == 'states_id' && $config['states_id_default']) {
                     continue;
                 }
+                //bypass for entities_id // default inventory conf is 0 'root entity')
+                if ($lock == 'entities_id') {
```

Comment:
```suggestion
                if ($lock == 'entities_id' && $config['entities_id_default'])) {
```

You assume that default entity is "0", but that's not true, since there is a parameter for default entity in inventory configuration.

Reply:
yes and default value is ```root entity``` => ```0```

Reply:
so when default value is ```0```, ```entities_id``` key is removed from ```input```

Reply:
Yes, indeed!
My point was the "inconsistence" with state; but that one can be nullable. Either apply my suggestion or add a comment, to make things clear ;)

Reply:
(anyways that's not a blocker ;))

Reply:
`&& $config['entities_id_default']` will anyway be evaluated to false if config defines root entity as default.